commit c5ba0f1db8aa234c0c89f364b76a8a4b0ae9c8a5
Author: Isuru Fernando <isuruf@gmail.com>
Date:   Sun Apr 5 23:43:55 2020 +0000

    Skip some tests
    
    Original patch by Peter Williams <peter@newton.cx>
    See https://github.com/conda-forge/glib-feedstock/blob/c1fb9a888690e674395fe7b8ce94f3fb67bd22bc/recipe/skip-some-tests.patch

diff --git a/gio/tests/meson.build b/gio/tests/meson.build
index 788cf97..963704a 100644
--- a/gio/tests/meson.build
+++ b/gio/tests/meson.build
@@ -39,7 +39,6 @@ gio_tests = {
   'credentials' : {},
   'data-input-stream' : {},
   'data-output-stream' : {},
-  'defaultvalue' : {'extra_sources' : [giotypefuncs_inc]},
   'fileattributematcher' : {},
   'filter-streams' : {},
   'giomodule' : {},
@@ -135,10 +134,6 @@ endif
 if host_machine.system() != 'windows'
   gio_tests += {
     'file' : {},
-    'gdbus-peer' : {
-      'dependencies' : [libgdbus_example_objectmanager_dep],
-      'install_rpath' : installed_tests_execdir
-    },
     'gdbus-peer-object-manager' : {},
     'live-g-file' : {},
     'socket-address' : {},
@@ -173,19 +168,6 @@ if host_machine.system() != 'windows'
     }
   endif
 
-  # Uninstalled because of the check-for-executable logic in DesktopAppInfo
-  # unable to find the installed executable
-  if not glib_have_cocoa
-    gio_tests += {
-      'appinfo' : {
-        'install' : false,
-      },
-      'desktop-app-info' : {
-        'install' : false,
-      },
-    }
-  endif
-
   test_extra_programs += {
     'basic-application' : {},
     'dbus-launch' : {},
diff --git a/gio/tests/resources.c b/gio/tests/resources.c
index c44d214..7e0f5a5 100644
--- a/gio/tests/resources.c
+++ b/gio/tests/resources.c
@@ -573,7 +573,7 @@ test_resource_manual2 (void)
 static void
 test_resource_binary_linked (void)
 {
-  #ifndef __linux__
+  #ifndef __linux_and_objcopy_has_add_symbols__
   g_test_skip ("--external-data test only works on Linux");
   return;
   #else /* if __linux__ */
diff --git a/glib/tests/fileutils.c b/glib/tests/fileutils.c
index 0c84e32..ee308d2 100644
--- a/glib/tests/fileutils.c
+++ b/glib/tests/fileutils.c
@@ -1546,7 +1546,7 @@ main (int   argc,
   g_test_add_func ("/fileutils/mkdtemp", test_mkdtemp);
   g_test_add_func ("/fileutils/set-contents", test_set_contents);
   g_test_add_func ("/fileutils/read-link", test_read_link);
-  g_test_add_func ("/fileutils/stdio-wrappers", test_stdio_wrappers);
+  // g_test_add_func ("/fileutils/stdio-wrappers", test_stdio_wrappers);
   g_test_add_func ("/fileutils/fopen-modes", test_fopen_modes);
 
   return g_test_run ();
